src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,34 @@
+package duke.command;
+
+import duke.main.Ui;
+import duke.utils.Parser;
+import duke.utils.Storage;
+import duke.utils.TaskList;
+
+import java.io.IOException;
+
+public class AddCommand implements Command {
+
+    @Override
+    public void execute(String task, Ui ui, Storage storage, TaskList taskList) {
+        String[] token = task.split(" ", 2);
+        if (token.length < 2) {
+            ui.setResponse("Adding task failed, task body cannot be empty");
```

Comment:
Alternatively, one could throw an exception and allow the Ui object to catch it. In this case, we can create an exception like AddTaskFailureException and give an appropriate message along with it.

In the Ui class, we can catch all the different exceptions thrown and apply Polymorphism to print the exception messages specific to each class. This could possibly make the overall code even neater.